hardware/detect.py

```diff
@@ -958,65 +984,18 @@ def _main(options):
     parse_dmesg(hrdw, output)
     bios_hp.dump_hp_bios(hrdw)
 
-    if "benchmark_cpu" in options:
-        bm_cpu.cpu_perf(hrdw)
-
-    if "benchmark_mem" in options:
-        bm_mem.mem_perf(hrdw)
-
-    if "benchmark_disk" in options:
-        bm_disk.disk_perf(hrdw,
-                          destructive=options['benchmark_disk_destructive'])
+    if parser.benchmark:
+        if 'cpu' in parser.benchmark:
```

Comment:
This feels strange ... you've limited parser.benchmark via choices so it will be a simple string and it *must* be one of the 3 options in that list.  so using `in` against a string will look for substrings which isn't really right why not just use a simple equaly test?

Reply:
parser.benchmark is actually a list and it allows multiple choices
just an example:
```
sudo ./hardware-detect --benchmark cpu mem disk

CPU Performance: 1 logical CPU to test (ETA: 20 seconds)
Benchmarking all CPUs for 10 seconds (8 threads)

Benchmarking memory @128M from all CPUs for 5 seconds (8 threads)

Running storage bench on 0 disks in non destructive mode for 0 seconds
```